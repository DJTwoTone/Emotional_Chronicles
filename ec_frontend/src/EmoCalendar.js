//https://openbase.com/js/react-calendar


import React, { useState } from 'react';
import Calendar from 'react-calendar';
import 'react-calendar/dist/Calendar.css'

function EmoCalendar () {


    //can i change this to [date, setDate] ?
    const [calDate, setCalDate] = useState(new Date())

    function showEntry(value, event) {
        console.log(typeof(value), value.getDate());
        //
    }
    console.log(calDate.getMonth())

    //make a useEffect to redirect to a day

    //the css needs to be adjusted to be bigger

    return (
        <div>
        <Calendar
            onChange={setCalDate}
            value={calDate}
            onClickDay={showEntry}
        />
        {/* <p>{{calDate}}</p> */}
        </div>
            )
}

export default EmoCalendar;