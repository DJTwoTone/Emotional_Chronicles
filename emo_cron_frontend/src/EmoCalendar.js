import React, { useContext, useState } from 'react';
import { DateTime } from 'luxon'

// import 'react-calendar/dist/Calendar.css';
import './Calendar.css';

import Calendar from 'react-calendar';

import Container from 'react-bootstrap/Container';

import MonthGraph from './MonthGraph';
import UserContext from './UserContext';

function EmoCalendar () {

    const { loggedInUser } = useContext(UserContext);

    const [calDate, setCalDate] = useState(new Date());
    const [firstOfTheMonth, setFirstOfTheMonth] =useState(calDate)

    function onChange(value, evt) {
        console.log('nextvalue', value, evt)
        setCalDate(value);
    }

    function onActiveStartDateChange({activeStartDate}) {
        
        setFirstOfTheMonth(activeStartDate)
    }
    //this need deigned
    //possibly an emotion cloud as the background
    
    console.log('calDAte', calDate, typeof(calDate))

    //use onDayClick (value, event) to go to diary entry
    //tileContent

    return (
        <Container className='m-4'>
        <Calendar 
            onChange={onChange}
            onActiveStartDateChange={onActiveStartDateChange}
            value={calDate}
            maxDetail='month'
            minDetail='month'

        />
        <MonthGraph date={DateTime.fromJSDate(firstOfTheMonth)}/>
        </Container>
            )
}

export default EmoCalendar;