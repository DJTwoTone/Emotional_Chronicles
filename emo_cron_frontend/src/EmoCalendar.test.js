import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import EmoCalendar from './EmoCalendar';

it('should render', function() {
    render(
        <MemoryRouter>
            <EmoCalendar />
        </MemoryRouter>
    );
})

it('should match the snapshot', function() {
    const { asFragment } = render(
        <MemoryRouter>
            <EmoCalendar />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
})