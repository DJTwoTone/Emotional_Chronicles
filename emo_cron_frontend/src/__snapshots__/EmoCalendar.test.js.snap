// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="justify-content-center my-5 container"
  >
    <div
      class="react-calendar"
    >
      <div
        class="react-calendar__navigation"
        style="display: flex;"
      >
        <button
          aria-label=""
          class="react-calendar__navigation__arrow react-calendar__navigation__prev2-button"
          type="button"
        >
          «
        </button>
        <button
          aria-label=""
          class="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
          type="button"
        >
          ‹
        </button>
        <button
          aria-label=""
          class="react-calendar__navigation__label"
          disabled=""
          style="flex-grow: 1;"
          type="button"
        >
          <span
            class="react-calendar__navigation__label__labelText react-calendar__navigation__label__labelText--from"
          >
            April 2021
          </span>
        </button>
        <button
          aria-label=""
          class="react-calendar__navigation__arrow react-calendar__navigation__next-button"
          type="button"
        >
          ›
        </button>
        <button
          aria-label=""
          class="react-calendar__navigation__arrow react-calendar__navigation__next2-button"
          type="button"
        >
          »
        </button>
      </div>
      <div
        class="react-calendar__viewContainer"
      >
        <div
          class="react-calendar__month-view "
        >
          <div
            style="display: flex; align-items: flex-end;"
          >
            <div
              style="flex-grow: 1; width: 100%;"
            >
              <div
                class="react-calendar__month-view__weekdays"
                style="display: flex; flex-wrap: no-wrap;"
              >
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                >
                  <abbr
                    aria-label="Sunday"
                    title="Sunday"
                  >
                    Sun
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                >
                  <abbr
                    aria-label="Monday"
                    title="Monday"
                  >
                    Mon
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                >
                  <abbr
                    aria-label="Tuesday"
                    title="Tuesday"
                  >
                    Tue
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                >
                  <abbr
                    aria-label="Wednesday"
                    title="Wednesday"
                  >
                    Wed
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                >
                  <abbr
                    aria-label="Thursday"
                    title="Thursday"
                  >
                    Thu
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                >
                  <abbr
                    aria-label="Friday"
                    title="Friday"
                  >
                    Fri
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                >
                  <abbr
                    aria-label="Saturday"
                    title="Saturday"
                  >
                    Sat
                  </abbr>
                </div>
              </div>
              <div
                class="react-calendar__month-view__days"
                style="display: flex; flex-wrap: wrap;"
              >
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden; margin-left: 57.142857142857146%;"
                  type="button"
                >
                  <abbr
                    aria-label="April 1, 2021"
                  >
                    1
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 2, 2021"
                  >
                    2
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 3, 2021"
                  >
                    3
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 4, 2021"
                  >
                    4
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 5, 2021"
                  >
                    5
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 6, 2021"
                  >
                    6
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 7, 2021"
                  >
                    7
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 8, 2021"
                  >
                    8
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 9, 2021"
                  >
                    9
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 10, 2021"
                  >
                    10
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 11, 2021"
                  >
                    11
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__tile--now react-calendar__tile--active react-calendar__tile--range react-calendar__tile--rangeStart react-calendar__tile--rangeEnd react-calendar__tile--rangeBothEnds react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 12, 2021"
                  >
                    12
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 13, 2021"
                  >
                    13
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 14, 2021"
                  >
                    14
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 15, 2021"
                  >
                    15
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 16, 2021"
                  >
                    16
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 17, 2021"
                  >
                    17
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 18, 2021"
                  >
                    18
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 19, 2021"
                  >
                    19
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 20, 2021"
                  >
                    20
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 21, 2021"
                  >
                    21
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 22, 2021"
                  >
                    22
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 23, 2021"
                  >
                    23
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 24, 2021"
                  >
                    24
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 25, 2021"
                  >
                    25
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 26, 2021"
                  >
                    26
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 27, 2021"
                  >
                    27
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 28, 2021"
                  >
                    28
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 29, 2021"
                  >
                    29
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  type="button"
                >
                  <abbr
                    aria-label="April 30, 2021"
                  >
                    30
                  </abbr>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <p>
        Month graph
      </p>
      <div
        class="recharts-responsive-container"
        style="width: 100%; height: 400px;"
      >
        <div
          height="400"
          width="0"
        >
          <div
            class="recharts-wrapper"
            style="position: relative; cursor: default; width: 500px; height: 250px;"
          >
            <svg
              class="recharts-surface"
              height="250"
              type="natural"
              version="1.1"
              viewBox="0 0 500 250"
              width="500"
            >
              <defs>
                <clippath
                  id="recharts14-clip"
                >
                  <rect
                    height="210"
                    width="390"
                    x="80"
                    y="5"
                  />
                </clippath>
              </defs>
              <g
                class="recharts-cartesian-grid"
              >
                <g
                  class="recharts-cartesian-grid-horizontal"
                >
                  <line
                    fill="none"
                    height="210"
                    offset="[object Object]"
                    stroke="#ccc"
                    stroke-dasharray="3 3"
                    width="390"
                    x="80"
                    x1="80"
                    x2="470"
                    y="5"
                    y1="5"
                    y2="5"
                  />
                  <line
                    fill="none"
                    height="210"
                    offset="[object Object]"
                    stroke="#ccc"
                    stroke-dasharray="3 3"
                    width="390"
                    x="80"
                    x1="80"
                    x2="470"
                    y="5"
                    y1="215"
                    y2="215"
                  />
                </g>
                <g
                  class="recharts-cartesian-grid-vertical"
                >
                  <line
                    fill="none"
                    height="210"
                    offset="[object Object]"
                    stroke="#ccc"
                    stroke-dasharray="3 3"
                    width="390"
                    x="80"
                    x1="80"
                    x2="80"
                    y="5"
                    y1="5"
                    y2="215"
                  />
                  <line
                    fill="none"
                    height="210"
                    offset="[object Object]"
                    stroke="#ccc"
                    stroke-dasharray="3 3"
                    width="390"
                    x="80"
                    x1="470"
                    x2="470"
                    y="5"
                    y1="5"
                    y2="215"
                  />
                </g>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
              >
                <line
                  class="recharts-cartesian-axis-line"
                  fill="none"
                  height="30"
                  orientation="bottom"
                  stroke="#666"
                  type="category"
                  width="390"
                  x="80"
                  x1="80"
                  x2="470"
                  y="215"
                  y1="215"
                  y2="215"
                />
                <g
                  class="recharts-cartesian-axis-ticks"
                >
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="30"
                      orientation="bottom"
                      stroke="#666"
                      type="category"
                      width="390"
                      x="80"
                      x1="80"
                      x2="80"
                      y="215"
                      y1="221"
                      y2="215"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="30"
                      orientation="bottom"
                      stroke="none"
                      text-anchor="middle"
                      type="category"
                      width="390"
                      x="80"
                      y="223"
                    >
                      <tspan
                        dy="0.71em"
                        x="80"
                      >
                        0
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="30"
                      orientation="bottom"
                      stroke="#666"
                      type="category"
                      width="390"
                      x="80"
                      x1="470"
                      x2="470"
                      y="215"
                      y1="221"
                      y2="215"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="30"
                      orientation="bottom"
                      stroke="none"
                      text-anchor="middle"
                      type="category"
                      width="390"
                      x="470"
                      y="223"
                    >
                      <tspan
                        dy="0.71em"
                        x="470"
                      >
                        auto
                      </tspan>
                    </text>
                  </g>
                </g>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
              >
                <line
                  class="recharts-cartesian-axis-line"
                  fill="none"
                  height="210"
                  orientation="left"
                  stroke="#666"
                  type="number"
                  width="60"
                  x="20"
                  x1="80"
                  x2="80"
                  y="5"
                  y1="5"
                  y2="215"
                />
                <g
                  class="recharts-cartesian-axis-ticks"
                />
              </g>
            </svg>
            <div
              class="recharts-legend-wrapper"
              style="position: absolute; width: 450px; height: auto; left: 20px; bottom: 5px;"
            >
              <ul
                class="recharts-default-legend"
                style="padding: 0px; margin: 0px; text-align: center;"
              >
                <li
                  class="recharts-legend-item legend-item-0"
                  style="display: inline-block; margin-right: 10px;"
                >
                  <svg
                    class="recharts-surface"
                    height="14"
                    style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                    version="1.1"
                    viewBox="0 0 32 32"
                    width="14"
                  >
                    <path
                      class="recharts-legend-icon"
                      d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
                      fill="none"
                      stroke="#DF4ADF"
                      stroke-width="4"
                    />
                  </svg>
                  <span
                    class="recharts-legend-item-text"
                    style="color: rgb(223, 74, 223);"
                  >
                    disgust
                  </span>
                </li>
                <li
                  class="recharts-legend-item legend-item-1"
                  style="display: inline-block; margin-right: 10px;"
                >
                  <svg
                    class="recharts-surface"
                    height="14"
                    style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                    version="1.1"
                    viewBox="0 0 32 32"
                    width="14"
                  >
                    <path
                      class="recharts-legend-icon"
                      d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
                      fill="none"
                      stroke="#008108"
                      stroke-width="4"
                    />
                  </svg>
                  <span
                    class="recharts-legend-item-text"
                    style="color: rgb(0, 129, 8);"
                  >
                    fear
                  </span>
                </li>
                <li
                  class="recharts-legend-item legend-item-2"
                  style="display: inline-block; margin-right: 10px;"
                >
                  <svg
                    class="recharts-surface"
                    height="14"
                    style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                    version="1.1"
                    viewBox="0 0 32 32"
                    width="14"
                  >
                    <path
                      class="recharts-legend-icon"
                      d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
                      fill="none"
                      stroke="#FFE953"
                      stroke-width="4"
                    />
                  </svg>
                  <span
                    class="recharts-legend-item-text"
                    style="color: rgb(255, 233, 83);"
                  >
                    joy
                  </span>
                </li>
                <li
                  class="recharts-legend-item legend-item-3"
                  style="display: inline-block; margin-right: 10px;"
                >
                  <svg
                    class="recharts-surface"
                    height="14"
                    style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                    version="1.1"
                    viewBox="0 0 32 32"
                    width="14"
                  >
                    <path
                      class="recharts-legend-icon"
                      d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
                      fill="none"
                      stroke="#5050FF"
                      stroke-width="4"
                    />
                  </svg>
                  <span
                    class="recharts-legend-item-text"
                    style="color: rgb(80, 80, 255);"
                  >
                    sadness
                  </span>
                </li>
                <li
                  class="recharts-legend-item legend-item-4"
                  style="display: inline-block; margin-right: 10px;"
                >
                  <svg
                    class="recharts-surface"
                    height="14"
                    style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                    version="1.1"
                    viewBox="0 0 32 32"
                    width="14"
                  >
                    <path
                      class="recharts-legend-icon"
                      d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
                      fill="none"
                      stroke="#018AE1"
                      stroke-width="4"
                    />
                  </svg>
                  <span
                    class="recharts-legend-item-text"
                    style="color: rgb(1, 138, 225);"
                  >
                    surprise
                  </span>
                </li>
                <li
                  class="recharts-legend-item legend-item-5"
                  style="display: inline-block; margin-right: 10px;"
                >
                  <svg
                    class="recharts-surface"
                    height="14"
                    style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                    version="1.1"
                    viewBox="0 0 32 32"
                    width="14"
                  >
                    <path
                      class="recharts-legend-icon"
                      d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
                      fill="none"
                      stroke="#00B510"
                      stroke-width="4"
                    />
                  </svg>
                  <span
                    class="recharts-legend-item-text"
                    style="color: rgb(0, 181, 16);"
                  >
                    trust
                  </span>
                </li>
              </ul>
            </div>
            <div
              class="recharts-tooltip-wrapper"
              style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
            >
              <div
                class="recharts-default-tooltip"
                style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
              >
                <p
                  class="recharts-tooltip-label"
                  style="margin: 0px;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <a
      href="/fullDiary/testuser"
    >
      <button
        class="btn btn-dark btn-block btn-lg"
        type="button"
      >
        Show Me All My Diary Entries
      </button>
    </a>
  </div>
</DocumentFragment>
`;
